fineract-provider/src/main/java/org/apache/fineract/portfolio/savings/service/SavingsAccountInterestPostingServiceImpl.java

```diff
@@ -434,7 +434,7 @@ protected void recalculateDailyBalances(final Money openingAccountBalance, final
                 }
 
                 runningBalance = runningBalance.plus(transactionAmount);
-                if (!transaction.getRunningBalance(transactionAmount.getCurrency()).isEqualTo(transactionAmount)) {
+                if (!transaction.getRunningBalance(transactionAmount.getCurrency()).isEqualTo(runningBalance)) {
```

Comment:
Nice catch! Can you please add a test case which covers this?

Reply:
No, this is a simple typo and a very edge case. I can not set up a use-case where the calculated runningBalance was changed, it is equal with the transaction amount, but does not equal with the running balance. Actually the whole condition could be removed.